py/redrock/rebin.py

```diff
@@ -125,24 +288,190 @@ def trapz_rebin(x, y, xnew=None, edges=None):
 
     return result
 
+def trapz_rebin_batch_gpu(x, y, xnew=None, edges=None, myz=None):
+    """Rebin y(x) flux density using trapezoidal integration between bin edges
+    GPU algorithm rebins in batch for all redshifts and bases and returns
+    3d array.
+
+    Notes:
+        y is interpreted as a density, as is the output, e.g.
+
+        >>> x = np.arange(10)
+        >>> y = np.ones(10)
+        >>> trapz_rebin(x, y, edges=[0,2,4,6,8])  #- density still 1, not 2
+        array([ 1.,  1.,  1.,  1.])
+
+    Args:
+        x (array): input x values.
+        y (array): input y values (for all bases).
+        edges (array): (optional) new bin edges.
+        myz (array): redshifts
```

Comment:
1. Ok - `x` and `edges` are 1-d just as in the CPU method.  `y` may be 1-d or 2-d.  I will update this.

2. Ok - will clarify - one major difference between CPU and GPU version is that on the GPU, multiplying `x` by `(1+z)` is done in the kernel so that it can be done in parallel.

3. Ok - base case is still valid but will add example of 2-d `y` array too.